### PR TITLE
feature(datasource/boss)

### PR DESCRIPTION
**Pull request by perlman**
This is a slightly reworked version of @alexbaden's Boss datasource. @neurodata has been using it heavily over the past year.

KeyCloak authentication is now via a pop-up rather than redirect, and the keycloak-js library is no longer used.

There are definitely a few things we still want to look at:
- promise warnings on the console while typing in hostnames
- attempt to do an immediate login (like Google OAuth2) in an iframe
- switch keycloak authentication workflow from implicit to authorization code

Sample URL: boss://https://api.boss.neurodata.io/collman/collman15v2/EM25K
(The data are public but a valid login is required and can be created on the KeyCloak auth page.)

### CODE REVIEW

**Comment by jbms** [comment]
Overall this looks great!

I was able to test that it worked.  The retry authorization button doesn't fully work: if you click retry while an auth tab is already open, it correctly closes the existing tab and opens a new one.  However, it shows a CANCELLED status message, and if you complete the auth in the new tab that was opened, the auth code doesn't recognize it.  However, if you then click retry again, it does work.  I noticed that the Brainmaps credential provider, that this one was based on, has the same issue with retry, though, so I'm fine with this issue not being fixed for now.

**Comment by alexbaden** [comment]
@googlebot This PR is ok with me.

**Comment by falkben** [comment]
Hi @jbms -- just checking in to see if there's anything else you need from us on this PR before merging.  Thanks for all your work on this - we love it!

File: src/neuroglancer/datasource/boss/frontend.ts [hunk 1]
```diff
@@ -0,0 +1,630 @@
...
+  channels: Map<string, ChannelInfo>;
+  scalingLevels: number;
+  coordFrameKey: string;
+  coordFrame?: CoordinateFrameInfo;
+  key: string;
+  collection: string;
+}
+
+function scaleVoxelSize(voxelSizeInput: vec3, voxelUnit: VoxelUnitType): vec3 {
+  switch (voxelUnit) {
```
[the remark this answers is not in the document]
Comment (perlman) [reply src/neuroglancer/datasource/boss/frontend.ts]: Thanks! Much cleaner.

**Comment by jbms** [comment]
The changes look good, thanks.  Let me know when they have been tested.

**Comment by falkben** [comment]
I have tested and can load BOSS data with these latest commits.  Here's a test URL: https://deploy-dev--viz-prod-neurodata.netlify.com/#!%7B%22layers%22:%7B%22bloss2018%22:%7B%22source%22:%22boss://https://api.boss.neurodata.io/bloss/bloss18/image%22%2C%22type%22:%22image%22%7D%7D%2C%22navigation%22:%7B%22pose%22:%7B%22position%22:%7B%22voxelSize%22:%5B3.890000104904175%2C3.890000104904175%2C50%5D%2C%22voxelCoordinates%22:%5B40993.6484375%2C53252.57421875%2C160%5D%7D%7D%2C%22zoomFactor%22:0.5965508371143481%7D%2C%22layout%22:%22xy%22%7D

Also working locally

**Pull request merged**